llmnrd.c

```diff
@@ -275,7 +292,7 @@ int main(int argc, char **argv)
 	if (llmnrd_sock_rtnl < 0)
 		goto out;
 
-	llmnr_init(hostname, ipv6);
+	llmnr_init((const char **) hostnames, name_count, ipv6);
```

Comment:
Can we avoid this cast somehow?

Reply:
The cast can be avoided if we have `char**` in the function signature of course, which I would not like that much. But I do not have any better way in mind for this C dialect.